### PR TITLE
migrate to netcoreapp3.1

### PR DESCRIPTION
**Pull request by brettfo**
`netcoreapp3.1` is the LTS, so we should congregate around that.

### CODE REVIEW

File: src/absil/ilreflect.fs [hunk 1]
```diff
@@ -1541,13 +1541,13 @@ let rec buildMethodPass2 cenv tref (typB: TypeBuilder) emEnv (mdef: ILMethodDef)
...
 
-#if !FX_RESHAPED_REFEMIT || NETCOREAPP3_0
+#if !FX_RESHAPED_REFEMIT || NETCOREAPP3_1
         // DefinePInvokeMethod was removed in early versions of coreclr, it was added back in NETCORE_APP3_0.
         // It has always been available in the desktop framework
         let methB = typB.DefinePInvokeMethod(mdef.Name, p.Where.Name, p.Name, attrs, cconv, rty, null, null, argtys, null, null, pcc, pcs)
 #else
-        // We currently build targeting netcoreapp2_1, and will continue to do so through this VS cycle
-        // but we can run on Netcoreapp3.0 so ... use reflection to invoke the api, when we are executing on netcoreapp3.0
+        // We currently build targeting netcoreapp3_1, and will continue to do so through this VS cycle
```
Comment (KevinRansom) [inline src/absil/ilreflect.fs:1549]: comment looks a bit wierd now.

**Comment by cartermp** [comment]
Noice. Unrelated, but what do you think of these lines?

https://github.com/dotnet/fsharp/blob/4dd07412884f4b7afca6fc1bbef923b31c95ff3b/global.json#L4-L13

Should some things be updated and/or removed?

File: src/absil/ilreflect.fs (hunk 1 above)
Comment (brettfo) [reply src/absil/ilreflect.fs:1549]: Updated.

**Comment by brettfo** [comment]
@cartermp Updated tool versions to `3.1.301` and `16.6` which are the current releases.  Arcade and Helix values are handled by the arcade updater.

**Pull request merged**